src/Core.UnitTests/EnvironmentVariableProviderTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarLint.VisualStudio.Integration.UnitTests;
+
+namespace SonarLint.VisualStudio.Core.UnitTests
+{
+    [TestClass]
+    public class EnvironmentVariableProviderTests
+    {
+        [TestMethod]
+        [DataRow(null)]
+        [DataRow("")]
+        public void TryGet_NoVariableName_ArgumentNullException(string variableName)
+        {
+            var testSubject = new EnvironmentVariableProvider();
+
+            Action act = () => testSubject.TryGet(variableName);
+
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("variableName");
+        }
+
+        [TestMethod]
+        public void TryGet_VariableDoesNotExist_Null()
+        {
+            var testSubject = new EnvironmentVariableProvider();
+
+            var result = testSubject.TryGet("test");
```

Comment:
It might be worth testing with a more-likely-to-be-unique environment variable name.